src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -3,18 +3,28 @@
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [clojure.tools.logging :as log]))
 
 (defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
+  [request jruby-service config puppet-version]
+  (try
+    (jruby/with-jruby-puppet jruby-puppet jruby-service
+                             (core/handle-request request jruby-puppet config))
+    (catch IllegalStateException e
+      (let [message (.getMessage e)]
+        (log/error message)
+        {:status  503
+         :body    message
+         :headers {"Content-Type"     "text/plain"
+                   "X-Puppet-version" puppet-version}}))))
```

Comment:
I think the "bad request" exceptions that we throw from `request-handler-core` aren't getting an `X-Puppet-Version` header appended in the response either.  Would be nice to see if we could come up with a more general solution that would add the `X-Puppet-Version` header into the response delivered from `handle-request` for any case that we didn't get one already in the response to the core Puppet request.  This would hopefully help us avoid having to add this in every time we come up with future responses that we have to produce at the Clojure layer.